app/models/scheduler.rb

```diff
@@ -0,0 +1,33 @@
+require './lib/modules/our_array_methods'
+class Scheduler
+  def initialize(user)
+    @user = user
+  end
+
+  def generate_schedule
+    @user.gardens.each do |garden|
+      garden.plants.each do |plant|
+        generate_plant_schedule(plant)
+      end
+    end
+  end
+
+  private
+
+  def generate_plant_schedule(plant)
+    clear_schedule(plant)
```

Comment:
So this is an interesting piece - if someone has customized their schedule, and a new schedule is generated, would that clear their customizations? 
Currently the user has to click a button to generate the schedule, but I think you were looking at refactoring to have this happen automatically. 
This could just be something to keep in mind when we get to customizing schedules.